File: primalTestDotNet/Entities/Trap.cs
namespace PrimalTestDotNet.Entities;

class Trap(Level.Level level, int x, int y) : IGameObject
{
    public char Sprite => 'c';

    public IntVector2 Position => new(x, y);

    public void Draw(EntityRenderer renderer)
    {
        renderer.Draw(this);
    }

    public void Interact(Hero hero)
    {
        level.RemoveGameObject(this);
        level.SetWall(Position.Y, Position.X);
        hero.JumpOverTrap();
    }
}

File: PrimalTestCpp/Trap.cpp
#include "Trap.h"
#include "EntityRenderer.h"
#include "Level.h"
#include "Hero.h"

Trap::Trap(Level& level, int x, int y) : _position(x, y), _level(level) {}

char Trap::getSprite() const {
	return 'c';
}

void Trap::interact(Hero& hero) {
	_level.setWall(_position.getX(), _position.getY());
	_level.removeGameObject(*this);
	hero.jumpOverTrap();
}

const IntVector2D& Trap::getPosition() const {
	return _position;
}

void Trap::draw(const EntityRenderer& entityRenderer) const {
	entityRenderer.draw(*this);
}